engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,6 +656,31 @@ protected void serviceStart() throws Exception
     }
   }
 
+  /**
+   * Modify config object by adding SSL related parameters into a resource for WebApp's use
+   *
+   * @param config  Configuration to be modified
+   */
+  private void addSSLConfigResource(Configuration config)
+  {
+    SSLConfig sslConfig = dag.getValue(Context.DAGContext.SSL_CONFIG);
+    if (sslConfig != null) {
+      String nodeLocalConfig = sslConfig.getConfigPath();
+      if (StringUtils.isNotEmpty(nodeLocalConfig)) {
+        config.addResource(new Path(nodeLocalConfig));
+      } else {
+        // create a configuration object and add it as a resource
+        Configuration sslConfigResource = new Configuration(false);
+        String keyStorePath = sslConfig.getKeyStorePath();  // need to strip out the root
+        keyStorePath = (new File(keyStorePath)).getName();
+        sslConfigResource.set(SSL_SERVER_KEYSTORE_LOCATION, new File(keyStorePath).getName(), Context.DAGContext.SSL_CONFIG.getLongName());
```

Comment:
double new File().getName(). Simplify to new Path(sslConfig.getKeyStorePath()).getName()